### ISSUE TITLE
Replace val_loss with monitor_metric

### ISSUE THREAD

**Issue by williamFalcon** [PyTorchLightning/pytorch-lightning#1866]
Validation epoch end has a special key called 'val_loss' which enables early stopping and checkpoint.

However, a better name is likely monitor_metric

```
    def validation_epoch_end(self, outputs):
        return {'monitor_metric': whatever_thing_i_want}
```

This makes it clear to the user that they can pass in anything (epoch, loss, accuracy, bleu, etc).

Obviously this change needs to be backward compatible
